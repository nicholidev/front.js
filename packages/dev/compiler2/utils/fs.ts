import { promises as fsp } from "fs";
import * as path from "path";

export async function writeFileSafe(
  file: string,
  contents: string
): Promise<string> {
  await fsp.mkdir(path.dirname(file), { recursive: true });
  await fsp.writeFile(file, contents);
  return file;
}

export async function writeFilesSafe(
  files: { file: string; contents: string }[]
): Promise<string[]> {
  return Promise.all(
    files.map(({ file, contents }) => writeFileSafe(file, contents))
  );
}

export async function createTemporaryDirectory(
  baseDir: string
): Promise<string> {
  return fsp.mkdtemp(path.join(baseDir, "remix-"));
}
