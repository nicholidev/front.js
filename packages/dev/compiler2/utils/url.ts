import * as path from "path";

export function createUrl(publicPath: string, file: string): string {
  return publicPath + file.split(path.win32.sep).join("/");
}
