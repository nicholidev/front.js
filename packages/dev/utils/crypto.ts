import type { BinaryLike } from "crypto";
import { createHash } from "crypto";

export function getHash(source: BinaryLike): string {
  return createHash("sha1").update(source).digest("hex");
}
