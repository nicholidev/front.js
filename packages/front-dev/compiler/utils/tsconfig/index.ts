import tsConfigPaths from "tsconfig-paths";

export function createMatchPath(tsconfigPath: string | undefined) {
  // There is no tsconfig to match paths against.
  if (!tsconfigPath) {
    return undefined;
  }

  // When passing a absolute path, loadConfig assumes that the path contains
  // a tsconfig file.
  // Ref.: https://github.com/dividab/tsconfig-paths/blob/v4.0.0/src/__tests__/config-loader.test.ts#L74
  let configLoaderResult = tsConfigPaths.loadConfig(tsconfigPath);

  if (configLoaderResult.resultType === "failed") {
    if (configLoaderResult.message === "Missing baseUrl in compilerOptions") {
      throw new Error(
        `ðŸš¨ Oops! No baseUrl found, please set compilerOptions.baseUrl in your tsconfig or jsconfig`
      );
    }
    return undefined;
  }

  return tsConfigPaths.createMatchPath(
    configLoaderResult.absoluteBaseUrl,
    configLoaderResult.paths,
    configLoaderResult.mainFields,
    configLoaderResult.addMatchAll
  );
}
