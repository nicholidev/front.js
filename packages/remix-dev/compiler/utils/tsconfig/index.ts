import tsConfigPaths from "tsconfig-paths";

import { loadTsConfig } from "./configLoader";
export { loadTsConfig } from "./configLoader";

export function createMatchPath() {
  let configLoaderResult = loadTsConfig();
  if (configLoaderResult.resultType === "failed") {
    return undefined;
  }

  let matchPath = tsConfigPaths.createMatchPath(
    configLoaderResult.absoluteBaseUrl,
    configLoaderResult.paths,
    configLoaderResult.mainFields,
    configLoaderResult.addMatchAll
  );

  return matchPath;
}
