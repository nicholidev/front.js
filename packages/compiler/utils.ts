import fs from "fs";
import path from "path";

export function readVendorFile(filename: string): string {
  let file = path.resolve(__dirname, "./vendor", filename);
  return fs.readFileSync(file, "utf-8");
}
