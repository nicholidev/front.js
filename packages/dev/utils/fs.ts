import { promises as fsp } from "fs";
import * as path from "path";
// @ts-expect-error
import readPackageJson from "read-package-json-fast";

export async function writeFileSafe(
  file: string,
  contents: string
): Promise<void> {
  await fsp.mkdir(path.dirname(file), { recursive: true });
  await fsp.writeFile(file, contents);
}

export async function createTemporaryDirectory(
  baseDir: string
): Promise<string> {
  return fsp.mkdtemp(path.join(baseDir, "remix-"));
}
