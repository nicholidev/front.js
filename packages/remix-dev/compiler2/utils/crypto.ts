import * as fs from "fs";
import type { BinaryLike } from "crypto";
import { createHash } from "crypto";

export function getHash(source: BinaryLike): string {
  return createHash("sha256").update(source).digest("hex");
}

export async function getFileHash(file: string): Promise<string> {
  return new Promise((accept, reject) => {
    let hash = createHash("sha256");
    fs.createReadStream(file)
      .on("error", error => reject(error))
      .on("data", data => hash.update(data))
      .on("close", () => {
        accept(hash.digest("hex"));
      });
  });
}
