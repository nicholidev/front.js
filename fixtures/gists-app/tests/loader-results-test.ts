import type { Browser, Page } from "puppeteer";
import puppeteer from "puppeteer";

import { getHtml } from "./utils";

const testPort = 3000;
const testServer = `http://localhost:${testPort}`;

describe("loader results", () => {
  let browser: Browser;
  let page: Page;
  beforeEach(async () => {
    browser = await puppeteer.launch();
    page = await browser.newPage();
  });

  afterEach(() => browser.close());

  describe("when a loader returns a redirect", () => {
    it("sends a redirect", async () => {
      let res = await page.goto(`${testServer}/gists/mjijackson`);
      let chain = res.request().redirectChain();
      expect(chain.length).toEqual(1);
      expect(chain[0].response().status()).toEqual(302);
      expect(chain[0].response().headers().location).toEqual("/gists/mjackson");
    });
  });

  describe("when a loader returns not found", () => {
    it("sends a not found page", async () => {
      let res = await page.goto(`${testServer}/gists/_why`);
      expect(res.status()).toEqual(404);
      expect(await getHtml(page, "[data-test-id=content]")).toMatchSnapshot();
    });
  });

  describe.skip("when the loader has an error", () => {
    it("sends a server error page", async () => {
      let res = await page.goto(`${testServer}/gists/DANGER`);
      expect(res.status()).toEqual(500);
      expect(await getHtml(page, "[data-test-id=content]")).toMatchSnapshot();
    });
  });
});
