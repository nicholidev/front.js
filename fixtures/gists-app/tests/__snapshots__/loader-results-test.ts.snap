// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader results when a loader returns not found sends a not found page 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <title>User _why not found</title>
    <meta name=\\"description\\" content=\\"View all of the gists from _why\\" />
    <link
      rel=\\"stylesheet\\"
      href=\\"//unpkg.com/@exampledev/new.css@1.1.3/new.css\\"
    />
    <link rel=\\"stylesheet\\" href=\\"/build/global-ec887178.css\\" />
    <link rel=\\"stylesheet\\" href=\\"/build/routes/gists-a6d2a823.css\\" />
  </head>
  <body class=\\"m-4\\">
    <div data-test-id=\\"/gists\\">
      <header>
        <h1>Gists</h1>
        <ul>
          <li>
            <a class=\\"text-blue-700 underline\\" href=\\"/gists/ryanflorence\\"
              >Ryan Florence<!-- -->
            </a>
          </li>
          <li>
            <a class=\\"text-blue-700 underline\\" href=\\"/gists/mjackson\\"
              >Michael Jackson<!-- -->
            </a>
          </li>
        </ul>
      </header>
      <div data-test-id=\\"/gists/$username\\">
        <h2>
          No gists for
          <!-- -->_why
        </h2>
      </div>
      <div>I am shared</div>
    </div>
    <script>
      window.__remixContext = {
        globalData: { enableScripts: true },
        manifest: {
          assets: {
            \\"entry-browser\\": { file: \\"entry-browser-284a45d2.js\\" },
            \\"global.css\\": { file: \\"global-ec887178.css\\" },
            \\"routes/gists\\": { file: \\"routes/gists-46a1683e.js\\" },
            \\"routes/gists/$username\\": {
              file: \\"routes/gists/$username-2db4e8d4.js\\",
            },
            \\"routes/gists.css\\": { file: \\"routes/gists-a6d2a823.css\\" },
          },
          routes: {
            \\"routes/gists\\": {
              id: \\"routes/gists\\",
              path: \\"gists\\",
              hasLoader: true,
            },
            \\"routes/gists/$username\\": {
              id: \\"routes/gists/$username\\",
              path: \\":username\\",
              hasLoader: true,
              parentId: \\"routes/gists\\",
            },
          },
          version: \\"73c7418a\\",
        },
        matches: [
          {
            params: {},
            pathname: \\"/gists\\",
            route: { id: \\"routes/gists\\", path: \\"gists\\", hasLoader: true },
          },
          {
            params: { username: \\"_why\\" },
            pathname: \\"/gists/_why\\",
            route: {
              id: \\"routes/gists/$username\\",
              path: \\":username\\",
              hasLoader: true,
              parentId: \\"routes/gists\\",
            },
          },
        ],
        publicPath: \\"/build/\\",
        routeData: {
          \\"routes/gists\\": {
            users: [
              { id: \\"ryanflorence\\", name: \\"Ryan Florence\\" },
              { id: \\"mjackson\\", name: \\"Michael Jackson\\" },
            ],
          },
          \\"routes/gists/$username\\": null,
        },
      };
    </script>
    <script type=\\"module\\">
      import * as route0 from \\"/build/routes/gists-46a1683e.js\\";
      import * as route1 from \\"/build/routes/gists/$username-2db4e8d4.js\\";
      window.__remixContext.routeModules = {
        \\"routes/gists\\": route0,
        \\"routes/gists/$username\\": route1,
      };
    </script>
    <script src=\\"/build/entry-browser-284a45d2.js\\" type=\\"module\\"></script>
  </body>
</html>
"
`;

exports[`loader results when the loader has an error sends a server error page 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <title>Error</title>
    <link
      rel=\\"stylesheet\\"
      href=\\"//unpkg.com/@exampledev/new.css@1.1.3/new.css\\"
    />
    <link rel=\\"stylesheet\\" href=\\"/build/global-ec887178.css\\" />
  </head>
  <body class=\\"m-4\\">
    <div data-test-id=\\"errors/500\\"><h1>500</h1></div>
    <script>
      window.__remixContext = {
        globalData: { enableScripts: true },
        manifest: {
          assets: {
            \\"entry-browser\\": { file: \\"entry-browser-284a45d2.js\\" },
            \\"global.css\\": { file: \\"global-ec887178.css\\" },
            \\"routes/500\\": { file: \\"routes/500-a7234038.js\\" },
          },
          routes: { \\"routes/500\\": { id: \\"routes/500\\", path: \\"/gists/DANGER\\" } },
          version: \\"73c7418a\\",
        },
        matches: [
          {
            params: {},
            pathname: \\"/gists/DANGER\\",
            route: { id: \\"routes/500\\", path: \\"/gists/DANGER\\" },
          },
        ],
        publicPath: \\"/build/\\",
        routeData: {
          \\"routes/500\\": {
            users: [
              { id: \\"ryanflorence\\", name: \\"Ryan Florence\\" },
              { id: \\"mjackson\\", name: \\"Michael Jackson\\" },
            ],
          },
        },
      };
    </script>
    <script type=\\"module\\">
      import * as route0 from \\"/build/routes/500-a7234038.js\\";
      window.__remixContext.routeModules = { \\"routes/500\\": route0 };
    </script>
    <script src=\\"/build/entry-browser-284a45d2.js\\" type=\\"module\\"></script>
  </body>
</html>
"
`;
